src/schedule-and-details/schedule-section/index.jsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+
+import { ScheduleRow, SCHEDULE_ROW_TYPES } from './schedule-row';
+import { CertificateDisplayRow } from './certificate-display-row';
+import messages from './messages';
+
+const ScheduleSection = ({
+  intl,
+  endDate,
+  startDate,
+  errorFields,
+  platformName,
+  enrollmentEnd,
+  enrollmentStart,
+  upgradeDeadline,
+  enrollmentEndEditable,
+  certificateAvailableDate,
+  certificatesDisplayBehavior,
+  onChange,
+}) => {
+  const enrollmentEndHelpText = intl.formatMessage(
+    messages.scheduleEnrollmentEndDateHelpText,
+  );
+  const enrollmentEndHelpTexRestricted = intl.formatMessage(
+    messages.scheduleEnrollmentEndDateRestrictedHelpText,
+    { platformName },
+  );
+  const computedEnrollmentEndHelpText = `${enrollmentEndHelpText} ${
+    enrollmentEndEditable ? enrollmentEndHelpTexRestricted : ''
+  }`;
+
+  const propsForScheduleFields = [
```

Comment:
I would also put these configurations in a separate file so as not to oversaturate the component with similar data

Reply:
I have a lot of dependencies for each object, so it would not be justified